protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -107,23 +110,38 @@ public T parse(InputStream stream) {
             }
           }
         }
+        CodedInputStream cis;
+        if (stream instanceof ReadableBuffer) {
+          // This is only a temporary solution, and deserves a better solution.  We copy here to
+          // avoid Protobuf's bad deserialization buffering.  In reality, we should be able to pass
+          // a collection of ByteBuffers to the decoder and avoid buffering at all.  The code below
+          // should eventually be removed in favor of a proper solution.
+          ReadableBuffer buffer = (ReadableBuffer) stream;
+          // copyBuffer must never leave scope and may never be modified.
+          ByteBuffer copyBuffer = ByteBuffer.allocate(buffer.readableBytes());
```

Comment:
ah nm, we need to copy since we're considering the backing buffer to be immutable (and it isn't)